modules/tinymce/src/plugins/wordcount/main/ts/core/Count.ts

```diff
@@ -14,11 +14,14 @@ import { getText } from './GetText';
 
 export type Counter = (node: Node, schema: Schema) => number;
 
+const removeZwsp = (text: string) =>
+  text.replace('\u200B', '');
+
 const strLen = (str: string): number =>
   str.replace(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g, '_').length;
 
 const countWords: Counter = (node: Node, schema: Schema): number => {
-  const text = getText(node, schema).join('\n');
+  const text = removeZwsp(getText(node, schema).join('\n'));
```

Comment:
Can we add a comment here as to why we're removing this here? Something like this maybe?
```suggestion
  // TINY-7484: The grapheme word boundary logic states a ZWSP (\u200B) should be treated as a word
  // boundary however word counting normally does not consider it as anything so we strip it out. 
  const text = removeZwsp(getText(node, schema).join('\n'));
```